Code/GraphMol/Descriptors/Wrap/rdMolDescriptors.cpp

```diff
@@ -1048,4 +1049,24 @@ BOOST_PYTHON_MODULE(rdMolDescriptors) {
   python::def("CalcNumBridgeheadAtoms", numBridgeheadAtoms,
               (python::arg("mol"), python::arg("atoms") = python::object()),
               docString.c_str());
+
```

Comment:
If you have a general way of doing this, that would be great.  I was unhappy with the canonical boost way of doing this:

http://www.boost.org/doc/libs/1_61_0/libs/python/doc/html/tutorial/tutorial/exposing.html#tutorial.exposing.class_virtual_functions

What I ended up doing is making a wrapped class that is an implementation of the virtual base that forwards the C++ calls to an underlying python object.

```
class PythonFilterMatch : public FilterMatcherBase {
  PyObject *functor;
  bool incref;

 public:
  PythonFilterMatch(PyObject *self)
      : FilterMatcherBase("Python Filter Matcher"),
        functor(self),
        incref(false){};

  // ONLY CALLED FROM C++ from the Clone operation
  PythonFilterMatch(const PythonFilterMatch &rhs)
      : FilterMatcherBase(rhs), functor(rhs.functor), incref(true) {
    python::incref(functor);
  }

  ~PythonFilterMatch() {
    if (incref) python::decref(functor);
  }
  virtual bool isValid() const {
    return python::call_method<bool>(functor, "IsValid");
  }

  virtual std::string getName() const {
    return python::call_method<std::string>(functor, "GetName");
  }
```

And then created a helper class that is easier to inherit from, i.e. has the whole api spelled out.  Here the "self" of this class is sent to the C++ object.  This is a bit more similar to what SWIG would do and it works pretty well.

```
class FilterMatcher(PythonFilterMatcher):
    def __init__(self, name="Unamed FilterMatcher"):
        self.name = name
        PythonFilterMatcher.__init__(self, self)

    def HasMatch(self, mol):
        """Return True if the filter matches the molecule"""
        raise NotImplementedError("Need to implement HasMatch(mol) in a subclass of %s",
                                  self.__class__.__name__)
```

However, if there is a more generic way, I'd be happy for it, this is a bit of boilerplate.